### PR TITLE
Fix versioning for v1.0.1 release

### PR DESCRIPTION
**Pull request by Nirusu**
Turns out I had to change the version plus the Helm index for a new release.

Also retagged the current container image as v1.0.0 and v1.0.1 for consistency.

@derpsteb Can you verify the Helm repo index works/looks correctly?

### CODE REVIEW

File: charts/edgeless/index.yaml [hunk 1]
```diff
@@ -1,6 +1,16 @@
 apiVersion: v1
 entries:
   azuredisk-csi-driver:
+  - apiVersion: v1
+    appVersion: v1.0.1
```
Comment (derpsteb) [inline charts/edgeless/index.yaml:5]: Quotes are recommended for this field.
https://helm.sh/docs/topics/charts/#the-chartyaml-file

File: charts/edgeless/latest/azuredisk-csi-driver/Chart.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v1
-appVersion: v1.0.0
+appVersion: v1.0.1
```
Comment (derpsteb) [inline charts/edgeless/latest/azuredisk-csi-driver/Chart.yaml:2]: same as above.

File: charts/edgeless/index.yaml (hunk 1 above)
Comment (Nirusu) [reply charts/edgeless/index.yaml:5]: It's directly the output of `helm repo index .` though...

File: charts/edgeless/v1.0.1/azuredisk-csi-driver/Chart.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v1
-appVersion: v1.0.0
+appVersion: v1.0.1
```
Comment (derpsteb) [inline charts/edgeless/v1.0.1/azuredisk-csi-driver/Chart.yaml:2]: same as above.

File: charts/edgeless/index.yaml (hunk 1 above)
Comment (Nirusu) [reply charts/edgeless/index.yaml]: Added nevertheless. Also formatted it for VS Code if we already make manual changes to it.

Comment (derpsteb) [reply charts/edgeless/index.yaml]: Yeah it's weird. Dunno what is the prefered way. Feel free to ignore :D. The page said something about parsing issues if you have specific appVersions.

**Comment by derpsteb** [comment]
Index worked fine for deploying to a cluster.

**Comment by Nirusu** [comment]
Fixed. YAML is shit. Don't get why Helm itself does not strictly define these types as strings if they recommend it in their docs. But I guess that's just common YAML tooling in Go.

Not going to fix that for the v1.0.0 release as I don't silently want to replace that. Didn't seem to be an issue before, though.

**Comment by daniel-weisse** [comment]
imo, if its output from a tool, we shouldnt manually modify it

**Comment by Nirusu** [comment]
Generally, I would agree. Reading the issues on the Helm repo it mainly seems to be an issue if you somehow have git commits included as they can parsed as numbers in scientific quotation.

The "v" should already implicate it's a string, and generally Helm also parses it as such. Would be a weird regression if it suddenly doesn't. So I think we should be good in any case at least for `appVersion` and don't need to do that?

So my suggestion would be to:
a) keep if for Chart.yaml since that's where Helm actually fixed this and a new `helm create` create will also at quotes
b) don't for index.yaml since it already passed a YAML parser / marshaller and is usually the file we always want to auto-generate due to the hash digests in there

If something magically breaks, complaints can be redirected to the Helm repository I guess since that seems to what their current version at least also does and seems to be intended as such.

@derpsteb Does that sound okay to you?

**Comment by derpsteb** [comment]
Absolutely fine with me. Didn't want to create a fuzz about it. Just saw it and remembered that I read about it.

**Review by derpsteb — APPROVED** [review]

File: charts/edgeless/index.yaml [hunk 4]
```diff
@@ -1,14 +1,26 @@
...
-    description: Azure disk Container Storage Interface (CSI) Storage Plugin with
-      on-node encryption support
-    digest: 7272ead1ee433ed3e77440901e1607701c955daab3161dc3e008ae2323858d0a
-    name: azuredisk-csi-driver
-    urls:
-    - v1.0.0/azuredisk-csi-driver.tgz
-    version: v1.0.0
-generated: "2022-09-28T11:13:35.770299627+02:00"
+    - apiVersion: v1
+      appVersion: "v1.0.1"
```
Comment (derpsteb) [inline charts/edgeless/index.yaml:5]: just for completeness: still has quotes :D
next version as well.

Comment (Nirusu) [reply charts/edgeless/index.yaml:5]: Haven't pushed changes yet ;)

**Review by daniel-weisse — APPROVED** [review]

**Pull request merged**